File: mvn-advent-of-code/src/test/java/advent/twenty_fifteen/Day09AllInASingleNightTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day09AllInASingleNightTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day09AllInASingleNight("/twenty_fifteen/Day09-AllInASingleNight-input.txt");

        assertEquals(207, dp.getPart1Answer().intValue());
        assertEquals(804, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day09AllInASingleNight("/twenty_fifteen/Day09-AllInASingleNight-example#1.txt");

        assertEquals(605, dp.getPart1Answer().intValue());
        assertEquals(982, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day09AllInASingleNightTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day09AllInASingleNightTest  extends FlatSpec with Matchers {
  "2015 Day 9 - Input" should "calculate answers" in {
    val dp = new Day09AllInASingleNight("/twenty_fifteen/Day09-AllInASingleNight-input.txt")

    dp.part1Answer should be(207)
    dp.part2Answer should be(804)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day09AllInASingleNight("/twenty_fifteen/Day09-AllInASingleNight-example#1.txt")

    dp.part1Answer should be(605)
    dp.part2Answer should be(982)
  }
}
